craft_parts/plugins/maven_plugin.py

```diff
@@ -74,6 +77,97 @@ def validate_environment(
                 reason=f"invalid maven version {version!r}",
             )
 
+        system_java_major_version = self._get_system_openjdk_java_major_version()
+        project_java_major_version = self._get_project_java_major_version()
+        # if the project Java version cannot be detected, let it try build anyway.
+        if project_java_major_version is None:
+            return
+
+        if system_java_major_version < project_java_major_version:
+            raise errors.PluginEnvironmentValidationError(
+                part_name=self._part_name,
+                reason="system Java version is less than project Java version."
+                f"system: {system_java_major_version}, project: {project_java_major_version}",
+            )
+
+    def _get_project_java_major_version(self) -> int | None:
+        options = cast(MavenPluginProperties, self._options)
+        maven_executable = "./mvnw" if options.maven_use_mvnw else "mvn"
+        effective_pom_path = Path("./effective.pom")
+        try:
+            self._execute(
+                # Write to file since writing to stdout without maven info logs require sudo, i.e.
+                # sudo mvn help:effective-pom -q -DforceStdout -doutput=/dev/stdout
+                f"{maven_executable} help:effective-pom -Doutput={effective_pom_path}"
+            )
```

Comment:
Hmm, I guess `_execute` doesn't return the output of the command? It is a bit confusing the `-doutput=/dev/stdout` since you wouldn't be able to capture that either? Perhaps the `_execute` method needs to be changed?

If it has to be written to a file, do yu think it should go into `tmp` or something? E.g., if it is written to the project directory, it could override some existing file or something

Reply:
Oh - perhaps I could explain the comment better.

What I meant was that: we need to somehow capture the output. One way for me to redirect the output would be using `-doutput=/dev/stdout`, which `self._execute()` call would capture. However, writing to `/dev/stdout` would require sudo access. Hence, we could just write to a file and read that instead.

However, in hindsight, I could also try redirecting the stderr to stdout with `2>&1`.

Reply:
Yeh this is quite unexpected, I think try `2>&1` and see if that helps you